all-rit/ALL
# 501 Story on Pre-Survey

# Balance grouping of Imagine MC pre-survey questions/responses.


How to test:
- Create new Postman POST request with JSON body (provided below as requestBody.txt), to endpoint "http://localhost:5005/imagine/presurvey".
- Open DB table imagine23 in DataGrip (table found in folder testing/public/tables).
- Attach debugger for backend + send request.
- Refresh imagine23 table to view newly populated row.
- Update the Postman request body with a new userID (like say test2) and send request again.
- Repeat to populate the table until you can clearly see user responses correctly balance between the 4 sections, "control", "experiential", "discomfortCountNonPOC" and "discomfortCountPOC" (see picture imagine23Populated)

![imagine23Populated](https://github.com/all-rit/ALL/assets/89709181/afcd482b-2eab-4987-bf36-c4541b67c1ce)
[requestBody.txt](https://github.com/all-rit/ALL/files/14438523/requestBody.txt)

Closes #501 


- [ ] No discrepancies across browsers. Eg. chrome vs safari
- [ ] Pages pass W3C Validation for HTML (https://validator.w3.org)
- [ ] CSS (https://jigsaw.w3.org/css-validator/)
- [ ] Google sign-in is operational
- [ ] Accessibility functions
- [ ] Pages can scale without distorting page
- [ ] No dead links
- [ ] Navbar is consistent across the site
- [ ] Pages are screen-reader accessible
- [ ] Contrast meets standards for accessibility
- [ ] Pages are keyboard accessible
- [ ] Passes WAVE Evaluation Tool chrome extension
